packages/visx-brush/src/BaseBrush.tsx

```diff
@@ -111,6 +125,148 @@ export default class BaseBrush extends React.Component<BaseBrushProps, BaseBrush
+        const { start, end, extent } = prevBrush;
+
+        start.x = Math.min(extent.x0, extent.x1);
+        start.y = Math.min(extent.y0, extent.y0);
+        end.x = Math.max(extent.x0, extent.x1);
+        end.y = Math.max(extent.y0, extent.y1);
+
+        return {
+          ...prevBrush,
+          activeHandle: null,
+          isBrushing: false,
+          brushingType: undefined,
+        };
+      });
+    }
+  };
+
+  handleMouseMove = (event: MouseEvent) => {
+    const { useWindowMoveEvents, onBrushEnd, resetOnEnd } = this.props;
+    const { brushingType, isBrushing, brushingOptions, start } = this.state;
+
+    if (!useWindowMoveEvents || !isBrushing) return;
+
+    const offsetX = event.pageX - (brushingOptions?.pageX || 0);
+    const offsetY = event.pageY - (brushingOptions?.pageY || 0);
+
+    if (['left', 'right', 'top', 'bottom'].includes(brushingType ?? '')) {
+      this.updateBrush((prevBrush: BaseBrushState) => {
+        const { x: x0, y: y0 } = prevBrush.start;
+        const { x: x1, y: y1 } = prevBrush.end;
+
+        return {
+          ...prevBrush,
+          isBrushing: true,
+          extent: {
+            ...prevBrush.extent,
+            ...this.getExtent(
+              {
+                x:
+                  brushingType === 'left'
+                    ? Math.min(Math.max(x0 + offsetX, prevBrush.bounds.x0), prevBrush.bounds.x1)
+                    : x0,
+                y:
+                  brushingType === 'bottom'
+                    ? Math.min(Math.max(y0 + offsetY, prevBrush.bounds.y0), prevBrush.bounds.y1)
+                    : y0,
+              },
+              {
+                x:
+                  brushingType === 'right'
+                    ? Math.min(Math.max(x1 + offsetX, prevBrush.bounds.x0), prevBrush.bounds.x1)
+                    : x1,
+                y:
+                  brushingType === 'bottom'
+                    ? Math.min(Math.max(y1 + offsetY, prevBrush.bounds.y0), prevBrush.bounds.y1)
+                    : y1,
+              },
+            ),
+          },
+        };
+      });
+    }
+
+    if (['move'].includes(brushingType ?? '')) {
+      this.updateBrush((prevBrush: BaseBrushState) => {
+        const { x: x0, y: y0 } = prevBrush.start;
+        const { x: x1, y: y1 } = prevBrush.end;
+        const validDx =
+          offsetX > 0
+            ? Math.min(offsetX, prevBrush.bounds.x1 - x1)
+            : Math.max(offsetX, prevBrush.bounds.x0 - x0);
+
+        const validDy =
+          offsetY > 0
+            ? Math.min(offsetY, prevBrush.bounds.y1 - y1)
+            : Math.max(offsetY, prevBrush.bounds.y0 - y0);
+
+        return {
+          ...prevBrush,
+          isBrushing: true,
+          extent: {
+            ...prevBrush.extent,
+            x0: x0 + validDx,
+            y0: y0 + validDy,
+            x1: x1 + validDx,
+            y1: y1 + validDy,
+          },
+        };
+      });
+    }
+
+    if (['select'].includes(brushingType ?? '')) {
```

Comment:
Could simplify a bit/avoid instantiating arrays

```suggestion
    if (brushingType === 'select') {
```